cores/cv32e40p/user_manual/source/debug.rst

```diff
@@ -70,6 +80,49 @@ Several trigger registers are required to adhere to specification. The following
 writes to Trigger registers are ignored and reads reflect CSR values.
 
+Debug state
+-----------
+
+As specified in `RISC-V Debug Specification <https://riscv.org/specifications/debug-specification/>`_ every hart that can be selected by
+the Debug Module is in exactly one of four states: ``nonexistent``, ``unavailable``, ``running`` or ``halted``. Harts are ``nonexistent`` if
+they will never be part of the system, no matter how long a user waits. Harts are ``unavailable`` if they might become available at a later
+time, or if there are other harts with higher indexes than this one. Harts may be ``unavailable`` for a variety of reasons including being
+reset or temporarily being powered down.  Harts are ``running`` when they are executing normally, as if no debugger was attached. This
+includes being in a low power mode or waiting for an interrupt, as long as a halt request will result in the hart being ``halted``.
+Harts are ``halted`` when they are in Debug Mode, only performing tasks on behalf of the debugger.
+
+The remainder of this section assumes that the CV32E40P will not be classified as ``nonexistent`` by the integrator.
+
+The CV32E40P signals to the Debug Module whether it is ``running`` or ``halted`` via its ``debug_running_o`` and ``debug_halted_o`` pins
+respectively. Therefore, assuming that this core will not be integrated as a ``nonexistent`` core, the CV32E40P is classified as ``unavailable``
+when neither ``debug_running_o`` or ``debug_halted_o`` is asserted. Upon ``rst_ni`` assertion the debug state will be ``unavailable`` until some
+cycle(s) after ``rst_ni`` has been deasserted and ``fetch_enable_i`` has been sampled high. After this point (until a next reset assertion) the
+core will transition between having its ``debug_halted_o`` or ``debug_running_o`` pin asserted depending whether the core is in debug mode or not.
+Exactly one of the ``debug_havereset_o``, ``debug_running_o``, ``debug_halted_o`` is asserted at all times.
+
+:numref:`debug-running` and show :numref:`debug-halted` show typical examples of transitioning into the ``running`` and ``halted`` states.
+
+.. figure:: ../images/debug_running.svg
+   :name: debug-running
+   :align: center
+   :alt:
+
+   Transition into debug ``running`` state
+
+.. figure:: ../images/debug_halted.svg
+   :name: debug-halted
+   :align: center
+   :alt:
+
+   Transition into debug ``halted`` state
+
+The key properties of the debug states are:
+
+ * The CV32E40P can remain in its ``unavailable`` state for an arbitrarily long time (depending on ``rst_ni`` and ``fetch_enable_i``).
+ * If ``debug_req_i`` is asserted after ``rst_ni`` deassertion and latest at the time of ``fetch_enable_i`` assertion, then the CV32E40P
+   is guaranteed to transition straight from its ``unavailable`` state into its ``halted`` state. If ``debug_req_i`` is asserted at a later
+   point in time, then the CV32E40P might transition through the ``running`` state on its ways to the ``halted`` state.
```

Comment:
consider replacing the last sentence with
"Otherwise, CV32E40P will transition into the ``running`` state."

Reply:
I will leave the text as is; your proposed replacement is not equivalent.